LiteCore/tests/VectorQueryTest.cc

```diff
@@ -680,4 +680,257 @@ N_WAY_TEST_CASE_METHOD(SIFTVectorQueryTest, "Inspect Vector Index", "[Query][.Ve
     CHECK(doc->asArray()->count() == 100);
 }
 
+TEST_CASE_METHOD(SIFTVectorQueryTest, "APPROX_VECTOR_DISTANCE Errors (Bad Metric)", "[.VectorSearch]") {
+    IndexSpec::VectorOptions opts =
+            IndexSpec::VectorOptions(128, vectorsearch::FlatClustering{256}, IndexSpec::DefaultEncoding);
+    string pre = R"(SELECT META(a).id FROM )"s + collectionName
+                 + R"( AS a ORDER BY APPROX_VECTOR_DISTANCE(a.vector, $target)";
+    string          post = R"() LIMIT 5)";
+    Retained<Query> query;
+
+    SECTION("Default Metric") {
+        // 'euclidean2 is the default metric
+        REQUIRE(opts.metric == vectorsearch::Metric::Euclidean2);
+        VectorQueryTest::createVectorIndex("vecIndex", "[ ['.vector'] ]", opts);
+
+        // specifying it in the VS function.
+        string queryStr = pre + ", 'euclidean2'" + post;
+        query           = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+        CHECK(query != nullptr);
+
+        query = nullptr;
+        {
+            ExpectingExceptions e;
+            queryStr = pre + ", 'cosine'" + post;
+            try {
+                query = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+            } catch ( error& err ) {
+                CHECK(err.domain == error::LiteCore);
+                CHECK(err.code == error::InvalidQuery);
+                CHECK("in 3rd argument to APPROX_VECTOR_DISTANCE, cosine does not match the index's metric, euclidean2"s
+                      == err.what());
+            }
+            CHECK(query == nullptr);
+        }
+    }
+
+    SECTION("Non-default Metric") {
```

Comment:
Do you think it it's worth to test all metrics we supported?

Reply:
It's good idea. At least there will be place all metrics are touched.
Will do it.